### PR TITLE
Use native generators/iterables, remove helper cruft

### PR DESCRIPTION
**Pull request by jakebailey**
This PR:

- Removes unused functions from `core.ts`.
- Removes `getEntries`, which with our new target was always equal to `Object.entries`
- Switches nearly every instance of iterators and iteration over to native generators / iterators and `for-of`, `yield`, `yield*`, etc.
- Switches a couple of `arrayFrom(it).map(fn)` to the more efficient but less discoverable `arrayFrom(it, fn)`.
- ~Removes `Collection` and `ReadonlyCollection` entirely, as they were only used in a couple of places which could be more specific.~ Split out into #52134 with #52133 for Push too.
- Removes `sum`, which was a weird function which arbitrarily added up values off of properties of objects.
- Adds new helper function variants that use iterators where convenient.

This PR doesn't remove `arrayFrom`, as it turns out to be much faster than `Array.from`. It also doesn't remove `isArray` pending #17002.

There's only one main instance where we don't use a native generator, and that's `decodeMappings`, which seems difficult to change.

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot perf test this

**Comment by jakebailey** [comment]
@typescript-bot perf test this

**Comment by jakebailey** [comment]
Running this through benchstat in its current state, it appears to all be noise and some wins that could be chance:

```
name                                   old s/op   new s/op   delta
Angular_v18_10_0_parseTime             3.98 ± 1%  3.96 ± 1%    ~     (p=0.699 n=6+6)
Angular_v18_10_0_bindTime              1.21 ± 2%  1.20 ± 2%    ~     (p=0.093 n=6+6)
Angular_v18_10_0_checkTime             8.60 ± 1%  8.59 ± 0%    ~     (p=0.699 n=6+6)
Angular_v18_10_0_emitTime              7.53 ± 0%  7.52 ± 0%    ~     (p=0.699 n=6+6)
Angular_v18_10_0_totalTime             21.3 ± 0%  21.3 ± 0%    ~     (p=0.180 n=6+6)
Angular_v16_17_1_parseTime             4.13 ± 0%  4.13 ± 1%    ~     (p=1.000 n=5+6)
Angular_v16_17_1_bindTime              1.26 ± 1%  1.27 ± 1%    ~     (p=0.329 n=6+5)
Angular_v16_17_1_checkTime             9.25 ± 0%  9.28 ± 1%    ~     (p=0.132 n=6+6)
Angular_v16_17_1_emitTime              7.97 ± 0%  7.98 ± 1%    ~     (p=0.485 n=6+6)
Angular_v16_17_1_totalTime             22.6 ± 0%  22.7 ± 1%    ~     (p=0.394 n=6+6)
Angular_v14_15_1_parseTime             4.18 ± 1%  4.14 ± 1%  -0.91%  (p=0.026 n=6+6)
Angular_v14_15_1_bindTime              1.30 ± 1%  1.30 ± 1%    ~     (p=0.855 n=6+6)
Angular_v14_15_1_checkTime             9.67 ± 1%  9.62 ± 1%    ~     (p=0.240 n=6+6)
Angular_v14_15_1_emitTime              8.23 ± 1%  8.16 ± 1%    ~     (p=0.093 n=6+6)
Angular_v14_15_1_totalTime             23.4 ± 1%  23.2 ± 0%    ~     (p=0.082 n=6+5)
Compiler_Unions_v18_10_0_parseTime     1.67 ± 1%  1.66 ± 1%    ~     (p=0.126 n=6+5)
Compiler_Unions_v18_10_0_bindTime      0.80 ± 0%  0.79 ± 1%  -0.78%  (p=0.000 n=5+6)
Compiler_Unions_v18_10_0_checkTime     9.47 ± 1%  9.40 ± 1%    ~     (p=0.084 n=6+6)
Compiler_Unions_v18_10_0_emitTime      2.78 ± 1%  2.77 ± 2%    ~     (p=0.818 n=6+6)
Compiler_Unions_v18_10_0_totalTime     14.7 ± 1%  14.6 ± 1%  -0.63%  (p=0.026 n=6+6)
Compiler_Unions_v16_17_1_parseTime     1.81 ± 0%  1.80 ± 1%  -0.68%  (p=0.009 n=6+6)
Compiler_Unions_v16_17_1_bindTime      0.84 ± 2%  0.84 ± 0%    ~     (p=0.126 n=6+5)
Compiler_Unions_v16_17_1_checkTime     10.2 ± 1%  10.1 ± 2%    ~     (p=0.394 n=6+6)
Compiler_Unions_v16_17_1_emitTime      3.09 ± 8%  2.98 ± 1%    ~     (p=0.485 n=6+6)
Compiler_Unions_v16_17_1_totalTime     15.9 ± 2%  15.8 ± 1%    ~     (p=0.180 n=6+6)
Compiler_Unions_v14_15_1_parseTime     1.85 ± 2%  1.84 ± 1%    ~     (p=0.180 n=6+6)
Compiler_Unions_v14_15_1_bindTime      0.88 ± 0%  0.87 ± 1%    ~     (p=0.114 n=4+6)
Compiler_Unions_v14_15_1_checkTime     10.1 ± 1%  10.2 ± 2%    ~     (p=0.180 n=6+6)
Compiler_Unions_v14_15_1_emitTime      3.13 ± 1%  3.23 ± 8%    ~     (p=0.589 n=6+6)
Compiler_Unions_v14_15_1_totalTime     16.0 ± 1%  16.1 ± 2%    ~     (p=0.394 n=6+6)
Monaco_v18_10_0_parseTime              2.95 ± 1%  2.95 ± 1%    ~     (p=0.905 n=6+6)
Monaco_v18_10_0_bindTime               1.05 ± 4%  1.04 ± 2%    ~     (p=0.699 n=6+6)
Monaco_v18_10_0_checkTime              7.25 ± 1%  7.20 ± 0%  -0.71%  (p=0.026 n=6+6)
Monaco_v18_10_0_emitTime               4.26 ± 1%  4.27 ± 1%    ~     (p=0.589 n=6+6)
Monaco_v18_10_0_totalTime              15.5 ± 1%  15.5 ± 0%    ~     (p=0.180 n=6+6)
Monaco_v16_17_1_parseTime              3.16 ± 1%  3.15 ± 1%    ~     (p=0.589 n=6+6)
Monaco_v16_17_1_bindTime               1.12 ± 1%  1.13 ± 1%    ~     (p=0.662 n=5+6)
Monaco_v16_17_1_checkTime              7.86 ± 1%  7.83 ± 0%    ~     (p=0.177 n=6+5)
Monaco_v16_17_1_emitTime               4.49 ± 1%  4.47 ± 1%    ~     (p=0.485 n=6+6)
Monaco_v16_17_1_totalTime              16.6 ± 1%  16.6 ± 1%    ~     (p=0.180 n=6+6)
Monaco_v14_15_1_parseTime              3.22 ± 0%  3.26 ± 3%    ~     (p=0.114 n=4+6)
Monaco_v14_15_1_bindTime               1.12 ± 2%  1.12 ± 2%    ~     (p=0.310 n=6+6)
Monaco_v14_15_1_checkTime              8.16 ± 1%  8.14 ± 0%    ~     (p=0.818 n=6+6)
Monaco_v14_15_1_emitTime               4.72 ± 1%  4.68 ± 1%  -0.85%  (p=0.015 n=6+6)
Monaco_v14_15_1_totalTime              17.2 ± 0%  17.2 ± 1%    ~     (p=0.240 n=6+6)
TFS_v18_10_0_parseTime                 2.25 ± 2%  2.25 ± 1%    ~     (p=0.937 n=6+6)
TFS_v18_10_0_bindTime                  1.17 ± 2%  1.16 ± 1%    ~     (p=0.589 n=6+6)
TFS_v18_10_0_checkTime                 6.84 ± 1%  6.78 ± 1%  -0.90%  (p=0.026 n=6+6)
TFS_v18_10_0_emitTime                  3.93 ± 1%  3.91 ± 1%    ~     (p=0.082 n=6+5)
TFS_v18_10_0_totalTime                 14.2 ± 1%  14.1 ± 0%  -0.73%  (p=0.004 n=6+5)
TFS_v16_17_1_parseTime                 2.59 ± 1%  2.59 ± 1%    ~     (p=0.818 n=6+6)
TFS_v16_17_1_bindTime                  1.05 ± 2%  1.07 ± 3%    ~     (p=0.180 n=6+6)
TFS_v16_17_1_checkTime                 7.39 ± 0%  7.42 ± 0%    ~     (p=0.093 n=6+6)
TFS_v16_17_1_emitTime                  4.21 ± 2%  4.21 ± 1%    ~     (p=0.818 n=6+6)
TFS_v16_17_1_totalTime                 15.2 ± 0%  15.3 ± 0%    ~     (p=0.093 n=6+6)
TFS_v14_15_1_parseTime                 2.73 ± 2%  2.70 ± 1%    ~     (p=0.132 n=6+6)
TFS_v14_15_1_bindTime                  1.08 ± 1%  1.08 ± 1%    ~     (p=0.727 n=6+6)
TFS_v14_15_1_checkTime                 7.72 ± 0%  7.72 ± 1%    ~     (p=0.699 n=6+6)
TFS_v14_15_1_emitTime                  4.58 ± 1%  4.54 ± 1%    ~     (p=0.126 n=6+5)
TFS_v14_15_1_totalTime                 16.1 ± 1%  16.1 ± 1%    ~     (p=0.394 n=6+6)
material_ui_v18_10_0_parseTime         3.52 ± 1%  3.50 ± 0%    ~     (p=0.052 n=6+5)
material_ui_v18_10_0_bindTime          0.96 ± 1%  0.96 ± 1%    ~     (p=0.699 n=6+6)
material_ui_v18_10_0_checkTime         17.0 ± 1%  16.9 ± 0%    ~     (p=0.257 n=6+4)
material_ui_v18_10_0_emitTime          0.00       0.00         ~     (all equal)
material_ui_v18_10_0_totalTime         21.5 ± 1%  21.4 ± 0%    ~     (p=0.476 n=6+4)
material_ui_v16_17_1_parseTime         3.66 ± 0%  3.67 ± 1%    ~     (p=0.429 n=5+6)
material_ui_v16_17_1_bindTime          1.02 ± 1%  1.01 ± 1%    ~     (p=0.065 n=6+6)
material_ui_v16_17_1_checkTime         18.0 ± 1%  17.9 ± 1%    ~     (p=0.065 n=6+6)
material_ui_v16_17_1_emitTime          0.00       0.00         ~     (all equal)
material_ui_v16_17_1_totalTime         22.7 ± 1%  22.6 ± 1%    ~     (p=0.180 n=6+6)
material_ui_v14_15_1_parseTime         3.89 ± 0%  3.88 ± 1%    ~     (p=0.792 n=5+6)
material_ui_v14_15_1_bindTime          1.05 ± 1%  1.05 ± 0%    ~     (p=0.177 n=6+5)
material_ui_v14_15_1_checkTime         19.0 ± 1%  19.0 ± 1%    ~     (p=0.485 n=6+6)
material_ui_v14_15_1_emitTime          0.00       0.00         ~     (all equal)
material_ui_v14_15_1_totalTime         23.9 ± 0%  23.9 ± 0%    ~     (p=0.662 n=6+5)
xstate_v18_10_0_parseTime              4.45 ± 1%  4.40 ± 0%  -1.17%  (p=0.016 n=5+4)
xstate_v18_10_0_bindTime               1.69 ± 1%  1.70 ± 1%    ~     (p=0.394 n=6+6)
xstate_v18_10_0_checkTime              2.62 ± 1%  2.60 ± 1%    ~     (p=0.485 n=6+6)
xstate_v18_10_0_emitTime               0.08 ± 0%  0.08 ± 0%    ~     (p=0.310 n=6+6)
xstate_v18_10_0_totalTime              8.84 ± 1%  8.79 ± 1%    ~     (p=0.310 n=6+6)
xstate_v16_17_1_parseTime              4.75 ± 0%  4.74 ± 1%    ~     (p=0.589 n=6+6)
xstate_v16_17_1_bindTime               1.66 ± 1%  1.66 ± 1%    ~     (p=0.429 n=6+5)
xstate_v16_17_1_checkTime              2.83 ± 1%  2.83 ± 1%    ~     (p=0.485 n=6+6)
xstate_v16_17_1_emitTime               0.09 ± 0%  0.09 ± 0%  -0.20%  (p=0.032 n=5+5)
xstate_v16_17_1_totalTime              9.32 ± 0%  9.29 ± 1%    ~     (p=0.065 n=6+6)
xstate_v14_15_1_parseTime              5.12 ± 1%  5.12 ± 1%    ~     (p=0.937 n=6+6)
xstate_v14_15_1_bindTime               1.59 ± 1%  1.59 ± 1%    ~     (p=0.818 n=6+6)
xstate_v14_15_1_checkTime              3.00 ± 0%  2.99 ± 1%    ~     (p=0.610 n=4+6)
xstate_v14_15_1_emitTime               0.10 ± 0%  0.10 ± 0%    ~     (p=0.288 n=6+6)
xstate_v14_15_1_totalTime              9.82 ± 0%  9.80 ± 1%    ~     (p=0.240 n=6+6)
```

File: src/compiler/core.ts [hunk 1]
```diff
@@ -2903,15 +2690,6 @@ export function enumerateInsertsAndDeletes<T, U>(newItems: readonly T[], oldItem
     return hasChanges;
 }
 
-/** @internal */
-export function fill<T>(length: number, cb: (index: number) => T): T[] {
```
Comment (jakebailey) [inline src/compiler/core.ts:2907]: This one's the same as `arrayOf`.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -347,8 +347,8 @@ import {
     hasAccessorModifier,
     hasAmbientModifier,
     hasContextSensitiveParameters,
-    hasDecorators,
     HasDecorators,
+    hasDecorators,
```
Comment (jakebailey) [inline src/compiler/checker.ts:351]: Annoying :(

**Comment by jakebailey** [comment]
@typescript-bot perf test this

**Comment by jakebailey** [comment]
This should be ready for review, though likely I can split out the Collection removal into a different PR. (The other PRs I've sent for cleanups had been pulled out of this one.)

**Comment by jakebailey** [comment]
Seems like it's still noise.

```
name                                   old s/op   new s/op   delta
Angular_v18_10_0_parseTime             3.98 ± 3%  3.98 ± 1%    ~     (p=0.485 n=6+6)
Angular_v18_10_0_bindTime              1.19 ± 1%  1.19 ± 1%    ~     (p=0.394 n=6+6)
Angular_v18_10_0_checkTime             8.58 ± 1%  8.63 ± 1%  +0.69%  (p=0.041 n=6+6)
Angular_v18_10_0_emitTime              7.53 ± 1%  7.57 ± 2%    ~     (p=0.485 n=6+6)
Angular_v18_10_0_totalTime             21.3 ± 0%  21.4 ± 1%    ~     (p=0.240 n=6+6)
Angular_v16_17_1_parseTime             4.17 ± 1%  4.15 ± 1%    ~     (p=0.310 n=6+6)
Angular_v16_17_1_bindTime              1.27 ± 1%  1.26 ± 0%  -0.71%  (p=0.045 n=6+6)
Angular_v16_17_1_checkTime             9.30 ± 1%  9.36 ± 1%    ~     (p=0.093 n=6+6)
Angular_v16_17_1_emitTime              7.97 ± 1%  7.97 ± 1%    ~     (p=1.000 n=6+6)
Angular_v16_17_1_totalTime             22.7 ± 1%  22.7 ± 1%    ~     (p=0.485 n=6+6)
Angular_v14_15_1_parseTime             4.16 ± 1%  4.17 ± 1%    ~     (p=1.000 n=6+6)
Angular_v14_15_1_bindTime              1.30 ± 1%  1.31 ± 1%    ~     (p=0.485 n=6+6)
Angular_v14_15_1_checkTime             9.67 ± 1%  9.68 ± 1%    ~     (p=0.853 n=6+6)
Angular_v14_15_1_emitTime              8.17 ± 0%  8.22 ± 1%    ~     (p=0.082 n=5+6)
Angular_v14_15_1_totalTime             23.3 ± 0%  23.3 ± 0%    ~     (p=0.792 n=6+5)
Compiler_Unions_v18_10_0_parseTime     1.66 ± 3%  1.65 ± 2%    ~     (p=0.485 n=6+6)
Compiler_Unions_v18_10_0_bindTime      0.80 ± 0%  0.79 ± 1%    ~     (p=0.117 n=5+6)
Compiler_Unions_v18_10_0_checkTime     9.48 ± 2%  9.50 ± 1%    ~     (p=0.937 n=6+6)
Compiler_Unions_v18_10_0_emitTime      2.77 ± 0%  2.78 ± 2%    ~     (p=0.537 n=5+6)
Compiler_Unions_v18_10_0_totalTime     14.7 ± 1%  14.7 ± 1%    ~     (p=1.000 n=6+6)
Compiler_Unions_v16_17_1_parseTime     1.81 ± 0%  1.80 ± 1%    ~     (p=0.065 n=6+6)
Compiler_Unions_v16_17_1_bindTime      0.84 ± 2%  0.84 ± 0%    ~     (p=0.171 n=6+4)
Compiler_Unions_v16_17_1_checkTime     10.1 ± 1%  10.1 ± 1%    ~     (p=0.310 n=6+6)
Compiler_Unions_v16_17_1_emitTime      2.99 ± 1%  3.07 ± 8%    ~     (p=0.394 n=6+6)
Compiler_Unions_v16_17_1_totalTime     15.8 ± 1%  15.8 ± 2%    ~     (p=0.589 n=6+6)
Compiler_Unions_v14_15_1_parseTime     1.85 ± 0%  1.85 ± 0%    ~     (p=0.753 n=6+5)
Compiler_Unions_v14_15_1_bindTime      0.88 ± 0%  0.87 ± 1%  -0.70%  (p=0.048 n=4+6)
Compiler_Unions_v14_15_1_checkTime     10.2 ± 1%  10.2 ± 1%    ~     (p=0.818 n=6+6)
Compiler_Unions_v14_15_1_emitTime      3.15 ± 2%  3.13 ± 1%    ~     (p=0.589 n=6+6)
Compiler_Unions_v14_15_1_totalTime     16.0 ± 1%  16.0 ± 1%    ~     (p=0.699 n=6+6)
Monaco_v18_10_0_parseTime              2.94 ± 2%  2.96 ± 0%    ~     (p=0.537 n=6+5)
Monaco_v18_10_0_bindTime               1.04 ± 2%  1.04 ± 2%    ~     (p=0.937 n=6+6)
Monaco_v18_10_0_checkTime              7.21 ± 0%  7.22 ± 0%    ~     (p=0.310 n=5+5)
Monaco_v18_10_0_emitTime               4.25 ± 0%  4.28 ± 2%    ~     (p=0.429 n=5+6)
Monaco_v18_10_0_totalTime              15.5 ± 1%  15.5 ± 0%    ~     (p=1.000 n=6+5)
Monaco_v16_17_1_parseTime              3.19 ± 2%  3.17 ± 2%    ~     (p=0.699 n=6+6)
Monaco_v16_17_1_bindTime               1.13 ± 2%  1.12 ± 2%    ~     (p=0.180 n=6+6)
Monaco_v16_17_1_checkTime              7.87 ± 1%  7.89 ± 1%    ~     (p=0.621 n=6+6)
Monaco_v16_17_1_emitTime               4.48 ± 1%  4.47 ± 1%    ~     (p=0.247 n=5+6)
Monaco_v16_17_1_totalTime              16.7 ± 1%  16.6 ± 1%    ~     (p=0.937 n=6+6)
Monaco_v14_15_1_parseTime              3.25 ± 2%  3.24 ± 1%    ~     (p=0.699 n=6+6)
Monaco_v14_15_1_bindTime               1.12 ± 1%  1.12 ± 1%    ~     (p=0.485 n=6+6)
Monaco_v14_15_1_checkTime              8.17 ± 1%  8.17 ± 0%    ~     (p=0.937 n=6+6)
Monaco_v14_15_1_emitTime               4.72 ± 1%  4.71 ± 1%    ~     (p=0.937 n=6+6)
Monaco_v14_15_1_totalTime              17.3 ± 1%  17.2 ± 0%    ~     (p=0.818 n=6+6)
TFS_v18_10_0_parseTime                 2.26 ± 1%  2.25 ± 1%    ~     (p=0.180 n=6+6)
TFS_v18_10_0_bindTime                  1.16 ± 1%  1.16 ± 2%    ~     (p=0.240 n=6+6)
TFS_v18_10_0_checkTime                 6.78 ± 1%  6.78 ± 1%    ~     (p=0.974 n=6+6)
TFS_v18_10_0_emitTime                  3.91 ± 1%  3.91 ± 1%    ~     (p=0.738 n=6+6)
TFS_v18_10_0_totalTime                 14.1 ± 1%  14.1 ± 0%    ~     (p=0.699 n=6+6)
TFS_v16_17_1_parseTime                 2.61 ± 3%  2.61 ± 2%    ~     (p=1.000 n=6+6)
TFS_v16_17_1_bindTime                  1.06 ± 3%  1.06 ± 2%    ~     (p=1.000 n=6+6)
TFS_v16_17_1_checkTime                 7.42 ± 1%  7.43 ± 1%    ~     (p=0.937 n=6+6)
TFS_v16_17_1_emitTime                  4.22 ± 1%  4.21 ± 1%    ~     (p=0.937 n=6+6)
TFS_v16_17_1_totalTime                 15.3 ± 1%  15.3 ± 1%    ~     (p=0.894 n=6+6)
TFS_v14_15_1_parseTime                 2.74 ± 1%  2.72 ± 2%    ~     (p=0.240 n=6+6)
TFS_v14_15_1_bindTime                  1.09 ± 2%  1.08 ± 0%    ~     (p=0.177 n=6+5)
TFS_v14_15_1_checkTime                 7.77 ± 0%  7.74 ± 1%    ~     (p=0.143 n=6+6)
TFS_v14_15_1_emitTime                  4.61 ± 2%  4.58 ± 1%    ~     (p=0.394 n=6+6)
TFS_v14_15_1_totalTime                 16.2 ± 0%  16.1 ± 1%  -0.51%  (p=0.026 n=6+6)
material_ui_v18_10_0_parseTime         3.53 ± 1%  3.54 ± 1%    ~     (p=0.818 n=6+6)
material_ui_v18_10_0_bindTime          0.97 ± 1%  0.96 ± 2%    ~     (p=0.420 n=6+6)
material_ui_v18_10_0_checkTime         17.0 ± 0%  16.9 ± 1%    ~     (p=0.082 n=5+6)
material_ui_v18_10_0_emitTime          0.00       0.00         ~     (all equal)
material_ui_v18_10_0_totalTime         21.5 ± 0%  21.4 ± 0%  -0.32%  (p=0.009 n=6+5)
material_ui_v16_17_1_parseTime         3.68 ± 1%  3.67 ± 1%    ~     (p=0.180 n=6+6)
material_ui_v16_17_1_bindTime          1.01 ± 1%  1.01 ± 1%    ~     (p=0.699 n=6+6)
material_ui_v16_17_1_checkTime         18.1 ± 2%  17.9 ± 1%    ~     (p=0.589 n=6+6)
material_ui_v16_17_1_emitTime          0.00       0.00         ~     (all equal)
material_ui_v16_17_1_totalTime         22.8 ± 1%  22.6 ± 1%    ~     (p=0.699 n=6+6)
material_ui_v14_15_1_parseTime         3.90 ± 1%  3.91 ± 1%    ~     (p=1.000 n=6+6)
material_ui_v14_15_1_bindTime          1.05 ± 1%  1.05 ± 2%    ~     (p=0.665 n=6+6)
material_ui_v14_15_1_checkTime         19.0 ± 1%  19.1 ± 1%    ~     (p=0.394 n=6+6)
material_ui_v14_15_1_emitTime          0.00       0.00         ~     (all equal)
material_ui_v14_15_1_totalTime         23.9 ± 1%  24.0 ± 1%    ~     (p=0.240 n=6+6)
xstate_v18_10_0_parseTime              4.43 ± 1%  4.43 ± 1%    ~     (p=0.937 n=6+6)
xstate_v18_10_0_bindTime               1.70 ± 1%  1.70 ± 1%    ~     (p=0.459 n=5+6)
xstate_v18_10_0_checkTime              2.60 ± 1%  2.61 ± 1%    ~     (p=0.699 n=6+6)
xstate_v18_10_0_emitTime               0.08 ± 0%  0.08 ± 0%    ~     (p=0.727 n=6+6)
xstate_v18_10_0_totalTime              8.81 ± 0%  8.81 ± 0%    ~     (p=0.485 n=6+6)
xstate_v16_17_1_parseTime              4.76 ± 0%  4.76 ± 0%    ~     (p=1.000 n=4+4)
xstate_v16_17_1_bindTime               1.66 ± 1%  1.66 ± 1%    ~     (p=0.699 n=6+6)
xstate_v16_17_1_checkTime              2.83 ± 0%  2.83 ± 1%    ~     (p=1.000 n=6+6)
xstate_v16_17_1_emitTime               0.09 ± 0%  0.09 ± 0%    ~     (p=0.429 n=5+6)
xstate_v16_17_1_totalTime              9.34 ± 0%  9.35 ± 0%    ~     (p=0.937 n=6+6)
xstate_v14_15_1_parseTime              5.14 ± 0%  5.12 ± 0%    ~     (p=0.095 n=5+5)
xstate_v14_15_1_bindTime               1.58 ± 0%  1.59 ± 1%    ~     (p=0.420 n=6+6)
xstate_v14_15_1_checkTime              3.00 ± 1%  2.99 ± 1%    ~     (p=0.589 n=6+6)
xstate_v14_15_1_emitTime               0.10 ± 0%  0.10 ± 0%    ~     (p=0.258 n=6+6)
xstate_v14_15_1_totalTime              9.81 ± 1%  9.80 ± 0%    ~     (p=0.937 n=6+6)
```

**Review by weswigham — APPROVED** [review]
The sourcemap stuff could *probably* be rewritten to be a generator, provided we'd be okay with some API changes.

**Comment by jakebailey** [comment]
> The sourcemap stuff could _probably_ be rewritten to be a generator, provided we'd be okay with some API changes.

Yeah, I just wasn't confident enough to attempt that in a cleanup PR; it seemed like a bunch of perf testing had gone into producing it, so, I'd want to make sure whatever it ends up as behaves how people expect.

**Pull request merged**